### PR TITLE
[Merged by Bors] - Allow specifying pvc parameters

### PR DESCRIPTION
**Pull request by maltesander**
# Description

- added pvc and resource parameters

closes: https://github.com/stackabletech/kafka-operator/issues/283

Integration Tests: https://github.com/stackabletech/integration-tests/pull/243

## Review Checklist

- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added (or not applicable)
- [ ] Documentation added (or not applicable)
- [ ] Changelog updated (or not applicable)
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
- [ ] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: rust/crd/src/lib.rs [hunk 1]
```diff
@@ -124,9 +212,18 @@ pub enum KafkaRole {
...
-pub struct KafkaConfig {}
+pub struct Storage {
+    #[serde(default)]
+    pub log_dirs: PvcConfig,
+}
+
+#[derive(Clone, Debug, Deserialize, JsonSchema, PartialEq, Serialize, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct KafkaConfig {
+    pub resources: Option<Resources<Storage, NoRuntimeLimits>>,
```
Comment (teozkr) [inline rust/crd/src/lib.rs:225]: I think this can just be `#[serde(default)]` without the `Option<T>`

Comment (teozkr) [reply rust/crd/src/lib.rs:225]: Shouldn't this be `JvmRuntimeLimits` btw?

Comment (maltesander) [reply rust/crd/src/lib.rs:225]: I would have assumed so, but i kept to how it was implemented in HDFS and ZooKeeper to be consistent. Does it even safe resources to specifiy a min value?

Comment (maltesander) [reply rust/crd/src/lib.rs]: done

**Review by teozkr — APPROVED** [review]

**Comment by maltesander** [comment]
bors merge

**Pull request closed without merging**